code/game/objects/structures/reflector.dm

```diff
@@ -30,6 +30,10 @@
 	P.current = reflector_turf
 	P.yo = reflect_turf.y - reflector_turf.y
 	P.xo = reflect_turf.x - reflector_turf.x
+	P.ignore_source_check = TRUE		//If shot by a laser, will now hit the mob that fired it
+	var/reflect_angle = dir2angle(new_dir)
+	P.setAngle(reflect_angle)

```

Comment:
This is all there was to make the Reflectors actually reflect

Reply:
simple solutions are the best usually. Good find